### PR TITLE
deps: Bump Rust nightly toolchain

### LINKED ISSUE #718: Bump toolchain to be on par with Rust 1.53.0
In accordance to our unofficial stability guidelines we aim to match the latest Rust stable release nightly toolchain. 1.53.0 was release recently on 2021-06-17 - this is a tracking issue for the progress of upgrading to that toolchain.

### PR DESCRIPTION
**Pull request by xla**
Closes #718

### CODE REVIEW

**Review by alexjg — DISMISSED** [review]

File: librad/src/git/p2p/transport.rs [hunk 1]
```diff
@@ -231,12 +232,15 @@ where
     io::Error::new(io::ErrorKind::Other, err)
 }
 
-fn block_on<F>(fut: F) -> io::Result<<F::Output as Try>::Ok>
+fn block_on<F>(fut: F) -> io::Result<<F::Output as Try>::Output>
 where
     F: Future,
-    F::Output: Try<Error = io::Error>,
+    F::Output: Try<Residual = io::Result<Infallible>>,
```
Comment (kim) [inline librad/src/git/p2p/transport.rs:238]: If this typechecks, the try-trait-v2 proposal is cleary unsound

Comment (kim) [reply librad/src/git/p2p/transport.rs:238]: Oh I get it. Not sure why they prefer to introduce such confusing terminology, where "Monad" would clearly do. I do not agree that this pulls it's weight here, feel free to specialise the type to `F: Future<Output = io::Result<T>>`.

Comment (FintanH) [reply librad/src/git/p2p/transport.rs:238]: Because Monads are scary

File: test/Cargo.toml [hunk 2]
```diff
@@ -20,7 +20,7 @@ either = "1"
 futures = ">= 0.3"
 futures_codec = "0.4"
 futures-await-test = "0"
-futures_ringbuf = "<= 0.3.0"
+futures_ringbuf = "= 0.3.1"
```
Comment (FintanH) [inline test/Cargo.toml:23]:
```suggestion
futures_ringbuf = "0.3"
```

File: test/src/test/integration/librad/scenario/collaboration.rs [hunk 3]
```diff
@@ -80,13 +80,13 @@ fn can_add_maintainer() {
                 move |storage| -> Result<Option<identities::VerifiedProject>, anyhow::Error> {
                     let project = identities::project::get(&storage, &rad)?.unwrap();
                     identities::project::update(
-                        &storage,
+                        storage,
                         &urn,
```
Comment (FintanH) [inline test/src/test/integration/librad/scenario/collaboration.rs:84]: Why doesn't `&urn` change to `urn` here? :thinking:

Comment (xla) [reply test/src/test/integration/librad/scenario/collaboration.rs:84]: AFAIU the closure that is passed to `using_storage` is of `FnOnce(&git::storage::Storage) -> A + Send + 'static` which means `storage` is is already a reference and before it read `&&storage` which is immediately dereferenced to `&storage` as per the expected signature of `identities::project::update`.

https://github.com/radicle-dev/radicle-link/blob/138813dba6d2cf377865ba163049982cb4c03c1c/librad/src/net/peer.rs#L248

https://github.com/radicle-dev/radicle-link/blob/138813dba6d2cf377865ba163049982cb4c03c1c/librad/src/git/identities/project.rs#L124-L130

Comment (FintanH) [reply test/src/test/integration/librad/scenario/collaboration.rs:84]: Ohhhh that makes sense :) Thanks for clearing that up!

File: librad/Cargo.toml [hunk 4]
```diff
@@ -83,8 +83,9 @@ version = "0.4"
 features = []
 
 [dependencies.git2]
-version = "=0.13.18"
+version = "=0.13.20"
 default-features = false
+# We add this feature to force git2 not to use the system libgit
```
Comment (FintanH) [inline librad/Cargo.toml:88]: What feature? :upside_down_face:

**Review by FintanH — APPROVED** [review]

File: librad/Cargo.toml (hunk 4 above)
Comment (xla) [reply librad/Cargo.toml:88]: Good one.

Comment (alexjg) [reply librad/Cargo.toml]: Comments are write only

**Review by FintanH — APPROVED** [review]
:kissing_closed_eyes::ok_hand:

**Review by alexjg — APPROVED** [review]

**Pull request merged**